save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityCollectionView.kt

```diff
@@ -128,21 +128,38 @@ val vulnerabilityCollectionView: FC<FossGraphCollectionViewProps> = FC { props -
                 th {
                     colSpan = tableInstance.visibleColumnsCount()
                     vulnerabilitiesFiltersRow {
-                        name = vulnerabilityFilters.prefixName
-                        onChangeFilters = { filterValue ->
-                            val filter = if (filterValue.isNullOrEmpty()) {
-                                VulnerabilityFilter("", vulnerabilityFilters.status)
-                            } else {
-                                VulnerabilityFilter(filterValue, vulnerabilityFilters.status)
-                            }
-                            setVulnerabilityFilters { filter }
-
-                            navigate(
-                                to = buildString {
+                        filter = vulnerabilityFilters
+                        onChangeFilter = { filterValue ->
+                            filterValue?.let {
+                                setVulnerabilityFilters(filterValue)
+                                val fullUrl = buildString {
                                     "/${FrontendRoutes.VULN}"
-                                    filterValue?.let { append("?vulnerabilityName=$filterValue") }
+                                    var isFirst = true
+
+                                    if (filterValue.prefixName.isNotBlank()) {
+                                        if (isFirst) {
+                                            append("?")
```

Comment:
@0x6675636b796f75676974687562 introduced http methods with typed parameters (`RequestUtils#254`) to avoid such manipulation. 
You can reuse it instead of checking `isFirst`